pyproject.toml

```diff
@@ -25,35 +36,61 @@ classifiers=[
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
 ]
-dependencies = [
-"numpy>=1.26.4,<2.0",
-"accelerate>=0.20.3,<0.40",
-"transformers>=4.34.1,<5.0,!=4.38.2",
-"torch>=2.2.0,<3.0",
-"sentencepiece>=0.1.99,<0.3",
-"tokenizers>=0.13.3,<1.0",
-"tqdm>=4.66.2,<5.0",
-"trl==0.8.6",
-"peft>=0.8.0,<0.13",
-"datasets>=2.15.0,<3.0",
-"fire>=0.5.0,<1.0",
-"simpleeval>=0.9.13,<1.0",
-]
 
-[project.optional-dependencies]
-dev = ["wheel>=0.42.0,<1.0", "packaging>=23.2,<24", "ninja>=1.11.1.1,<2.0", "scikit-learn>=1.0, <2.0", "boto3>=1.34, <2.0"]
-flash-attn = ["flash-attn>=2.5.3,<3.0"]
-aim = ["aim>=3.19.0,<4.0"]
-fms-accel = [
-  "fms_acceleration @ git+https://github.com/foundation-model-stack/fms-acceleration.git#subdirectory=plugins/framework"
-]
+packages= [{include="tuning"}, {include="launcher"}]
+
+
+[tool.poetry.group.dev]
+optional=true
+
+[tool.poetry.group.dev.dependencies]
+wheel=">=0.42.0,<1.0"
+packaging =">=23.2,<24"
+ninja=">=1.11.1.1,<2.0"
+scikit-learn=">=1.0,<2.0"
+boto3 = ">=1.34,<2.0"
+pytest = ">=7.0.0"
+build = "^1.0.0"
+setuptools = "^70.1.0"
+pre-commit = "^3.0.0"
+pylint = ">=2.16.2,<=3.1.0"
+coverage = "^7.5.4"
+genbadge = {extras = ["coverage"], version = "^1.1.1"}
+
+[tool.poetry.group.flash-attn]
+optional=true
+
+[tool.poetry.group.flash-attn.dependencies]
+flash-attn=">=2.5.3,<3.0"
+
+[tool.poetry.group.aim]
+optional=true
+
+[tool.poetry.group.aim.dependencies]
+aim=">=3.19.0,<4.0"
+
+
+[tool.poetry.group.fms-accel]
+optional=true
+
+[tool.poetry.group.fms-accel.dependencies]
+fms_acceleration = {git = "https://github.com/foundation-model-stack/fms-acceleration.git", subdirectory="plugins/framework", rev = "40aad4683e01be3faebbcaae2670524c431cbf3e"}
```

Comment:
Actually, this doesn't make sense, so I'll revert this change. Poetry already takes  care of "pinning" this dependency for the reproducible virtual environments it creates (i.e. the `poetry.lock` file).

We just have to run `poetry lock` (or manually update python dependencies) whenever fms_acceleration releases a new version and we'd like to update the dependencies of fms-hf-tuning to match those of fms_acceleration.

For example, at the moment fms_acceleration has a constraint [`transformers<4.40.0`](https://github.com/foundation-model-stack/fms-acceleration/blob/40aad4683e01be3faebbcaae2670524c431cbf3e/plugins/framework/pyproject.toml#L26) which propagates to fms-hf-tuning and sets transformers to 4.39.3.